payment/src/test/java/org/killbill/billing/payment/logging/SpyLogger.java

```diff
@@ -0,0 +1,671 @@
+package org.killbill.billing.payment.logging;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.Marker;
+import org.slf4j.helpers.FormattingTuple;
+import org.slf4j.helpers.MessageFormatter;
+
+import com.google.common.base.Optional;
+
+public class SpyLogger implements Logger {
```

Comment:
Could you leverage something like `SubstituteLogger` and delegate to the original logger? I'd really like to avoid copying the whole thing.

Alternatively, you could simply create a `org.slf4j.impl` package in the tests.